.github/workflows/scripts/scanner-fetch-nvd.sh

```diff
@@ -0,0 +1,79 @@
+#!/bin/bash
+set -euo pipefail
+
+base_url="https://services.nvd.nist.gov/rest/json/cves/2.0"
+dir="nvd-data"
+mkdir -p "$dir"
+
+info() { echo >&2 "$(date): INFO: $@"; }
+
+# Compression and upload
+compress_upload() {
+  tar -czf "$dir.tar.gz" -C "$dir" .
+  info "Compressed file size: $(stat -c %s "$dir.tar.gz") bytes"
+  mkdir -p "nvd-bundle" && mv "$dir.tar.gz" "nvd-bundle/$dir.tar.gz"
+  gsutil cp -r "nvd-bundle" "gs://scanner-v4-test/" || { info "Upload failed"; exit 1; }
+}
+
+dates() {
+  year=$1
+  echo "$year-01-01T00:00:00.000"
+  echo "$year-03-31T23:59:59.999"
+  echo "$year-04-01T00:00:00.000"
+  echo "$year-06-30T23:59:59.999"
+  echo "$year-07-01T00:00:00.000"
+  echo "$year-09-30T23:59:59.999"
+  echo "$year-10-01T00:00:00.000"
+  echo "$year-12-31T23:59:59.999"
+}
+
+fetch_data() {
+  nvd_key=$1
+
+  while read start && read end; do
+    info "Fetching pages from $start to $end"
+
+    local index=0
+    local total=0  # Initialize total as 0
+    local count
+    local temp_file
+
+    while (( index < total || total == 0 )); do
+      temp_file=$(mktemp)
+      info "Fetching page at index $index (out of ${total:-unknown total items})"
+
+      local request_url="${base_url}?noRejected&pubStartDate=$start&pubEndDate=$end&startIndex=$index"
+      if ! curl --get \
+                --fail \
+                --retry 3 \
+                --output "$temp_file" \
+                -H "apiKey: $nvd_key" \
+                "$request_url"; then
+            info "Failed to download page at index $index"
+            rm -f "$temp_file"
+            exit 1
+        fi
+
+      total=$(jq -r '.totalResults' "$temp_file")
+      count=$(jq -r '.resultsPerPage' "$temp_file")
+      jq '.vulnerabilities' "$temp_file"
+
+      rm -f "$temp_file"
+      index=$(( index + count ))
+
+      info "Fetching next page at $index (out of $total) in 2s"
```

Comment:
This will print a log line saying it will fetch the next page even if when index + count > total.